core/src/test/java/org/opensearch/sql/planner/physical/TrendlineOperatorTest.java

```diff
@@ -371,28 +407,452 @@ public void calculates_simple_moving_average_timestamp() {
 
+    List<ExprValue> result = execute(plan);
+    assertEquals(3, result.size());
+    assertThat(
+        result,
+        containsInAnyOrder(
+            tupleValue(ImmutableMap.of("timestamp", Instant.EPOCH)),
+            tupleValue(
+                ImmutableMap.of(
+                    "timestamp",
+                    Instant.EPOCH.plusMillis(1000),
+                    "timestamp_alias",
+                    Instant.EPOCH.plusMillis(500))),
+            tupleValue(
+                ImmutableMap.of(
+                    "timestamp",
+                    Instant.EPOCH.plusMillis(1500),
+                    "timestamp_alias",
+                    Instant.EPOCH.plusMillis(1250)))));
+  }
+
+  @Test
+  public void calculates_weighted_moving_average_one_field_one_sample() {
+    mockPlanWithData(List.of(tupleValue(ImmutableMap.of("distance", 100, "time", 10))));
+
+    var plan =
+        new TrendlineOperator(
+            inputPlan,
+            Collections.singletonList(
+                Pair.of(
+                    AstDSL.computation(1, AstDSL.field("distance"), "distance_alias", WMA),
+                    ExprCoreType.DOUBLE)));
+
     plan.open();
+
     assertTrue(plan.hasNext());
     assertEquals(
-        ExprValueUtils.tupleValue(ImmutableMap.of("timestamp", Instant.EPOCH)), plan.next());
-    assertTrue(plan.hasNext());
-    assertEquals(
-        ExprValueUtils.tupleValue(
-            ImmutableMap.of(
-                "timestamp",
-                Instant.EPOCH.plusMillis(1000),
-                "timestamp_alias",
-                Instant.EPOCH.plusMillis(500))),
-        plan.next());
-    assertTrue(plan.hasNext());
-    assertEquals(
-        ExprValueUtils.tupleValue(
-            ImmutableMap.of(
-                "timestamp",
-                Instant.EPOCH.plusMillis(1500),
-                "timestamp_alias",
-                Instant.EPOCH.plusMillis(1250))),
+        tupleValue(ImmutableMap.of("distance", 100, "time", 10, "distance_alias", 100)),
         plan.next());
-    assertFalse(plan.hasNext());
+  }
+
+  @Test
+  public void calculates_weighted_moving_average_one_field_two_samples() {
+    mockPlanWithData(
+        List.of(
+            tupleValue(ImmutableMap.of("distance", 100, "time", 10)),
+            tupleValue(ImmutableMap.of("distance", 200, "time", 10))));
+
+    var plan =
+        new TrendlineOperator(
+            inputPlan,
+            Collections.singletonList(
+                Pair.of(
+                    AstDSL.computation(2, AstDSL.field("distance"), "distance_alias", WMA),
+                    ExprCoreType.DOUBLE)));
+
+    List<ExprValue> result = execute(plan);
+    assertEquals(2, result.size());
+    assertThat(
+        result,
+        containsInAnyOrder(
+            tupleValue(ImmutableMap.of("distance", 100, "time", 10)),
+            tupleValue(
+                ImmutableMap.of(
+                    "distance", 200, "time", 10, "distance_alias", 166.66666666666663))));
```

Comment:
if you can, simplify this magic number. 
its hard to tell if this number is correct.